api/src/main/java/uk/gov/dhsc/htbhf/dwp/service/EligibilityService.java

```diff
@@ -3,23 +3,82 @@
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
+import uk.gov.dhsc.htbhf.dwp.entity.Child;
+import uk.gov.dhsc.htbhf.dwp.entity.legacy.LegacyHousehold;
+import uk.gov.dhsc.htbhf.dwp.entity.uc.UCHousehold;
 import uk.gov.dhsc.htbhf.dwp.model.EligibilityRequest;
 import uk.gov.dhsc.htbhf.dwp.model.EligibilityResponse;
+import uk.gov.dhsc.htbhf.dwp.repository.LegacyHouseholdRepository;
+import uk.gov.dhsc.htbhf.dwp.repository.UCHouseholdRepository;
+
+import java.time.LocalDate;
+import java.util.Optional;
+import java.util.Set;
 
 @Service
 public class EligibilityService {
 
     private static final String ENDPOINT = "/v1/dwp/benefits";
     private final String uri;
     private final RestTemplate restTemplate;
+    private final UCHouseholdRepository ucHouseholdRepository;
+    private final LegacyHouseholdRepository legacyHouseholdRepository;
 
-    public EligibilityService(@Value("${dwp.base-uri}") String baseUri, RestTemplate restTemplate) {
+    public EligibilityService(@Value("${dwp.base-uri}") String baseUri,
+                              RestTemplate restTemplate,
+                              UCHouseholdRepository ucHouseholdRepository,
+                              LegacyHouseholdRepository legacyHouseholdRepository) {
         this.uri = baseUri + ENDPOINT;
         this.restTemplate = restTemplate;
+        this.ucHouseholdRepository = ucHouseholdRepository;
+        this.legacyHouseholdRepository = legacyHouseholdRepository;
     }
 
     public EligibilityResponse checkEligibility(EligibilityRequest eligibilityRequest) {
```

Comment:
I think this method needs documenting, this is the crux of this Service, so let's at least details what the business rules are that are being applied here, i.e. why we're going to the database first and why we go for UC first.

Reply:
Added javadoc